fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -561,6 +565,28 @@ public static int getThreadsPerClientVersion() {
         return this.blockingInAsyncDetectionSupplier;
     }
 
+    /**
+     * Installs a listener that will be handed a collection of low-level I/O metrics that were gathered
+     * over the lifetime of a transaction. These metrics will be issued the listener as soon as a transaction
+     * is committed or closed. Because this listener is in the path of the completion of every transaction,
+     * implementations should take care to ensure that they are thread safe and efficiently process the provided
+     * metrics before returning.
+     *
+     * @param transactionMetricListener a consumer that accepts the database for which the transaction was
+     *   committed or closed as well as a {@code StoreTimer} containing the I/O metrics for the transaction
```

Comment:
This works, but it kind of feels like it's deserving of its own interface instead of just `BiConsumer`.

Might a listener want to know whether the transaction was committed or rolled back?

Reply:
Yeah, I was just recently thinking about something like that myself. Good suggestion. Done.